tests/realtikvtest/testkit.go

```diff
@@ -134,60 +140,71 @@ func Setup(t *testing.T, ks string, opts ...mockstore.MockTiKVStoreOption) (kv.S
 	var dom *domain.Domain
 	var err error
 
+	option := &realtikvStoreOption{}
+	for _, opt := range opts {
+		opt(option)
+	}
+	var ks string
+	if kerneltype.IsNextGen() {
+		if option.keyspace == "" {
+			// in nextgen kernel, SYSTEM keyspace must be bootstrapped first, if we
+			// don't specify a keyspace which normally is not specified, we use SYSTEM
+			// keyspace as default to make sure test cases can run correctly.
+			ks = keyspace.System
+		} else {
+			ks = option.keyspace
+		}
+		t.Log("create realtikv store with keyspace:", ks)
+	}
 	session.SetSchemaLease(500 * time.Millisecond)
 
-	if *WithRealTiKV {
-		path := *TiKVPath
-		if len(ks) > 0 {
-			path += "&keyspaceName=" + ks
-		}
-		var d driver.TiKVDriver
-		config.UpdateGlobal(func(conf *config.Config) {
-			conf.TxnLocalLatches.Enabled = false
-			conf.KeyspaceName = ks
-			conf.Store = config.StoreTypeTiKV
-		})
-		store, err = d.Open(path)
+	path := *TiKVPath
+	if len(ks) > 0 {
+		path += "&keyspaceName=" + ks
+	}
+	var d driver.TiKVDriver
+	config.UpdateGlobal(func(conf *config.Config) {
+		conf.TxnLocalLatches.Enabled = false
+		conf.KeyspaceName = ks
+		conf.Store = config.StoreTypeTiKV
+	})
+	store, err = d.Open(path)
+	require.NoError(t, err)
+	if kerneltype.IsNextGen() && ks != keyspace.System {
+		sysPath := *TiKVPath + "&keyspaceName=" + keyspace.System
+		sysStore, err := d.Open(sysPath)
 		require.NoError(t, err)
-		if kerneltype.IsNextGen() && ks != keyspace.System {
-			sysPath := *TiKVPath + "&keyspaceName=" + keyspace.System
-			sysStore, err := d.Open(sysPath)
-			require.NoError(t, err)
-			kvstore.SetSystemStorage(sysStore)
-			t.Cleanup(func() {
-				require.NoError(t, sysStore.Close())
-			})
+		kvstore.SetSystemStorage(sysStore)
+		t.Cleanup(func() {
+			require.NoError(t, sysStore.Close())
+		})
+	}
+	require.NoError(t, ddl.StartOwnerManager(context.Background(), store))
+	dom, err = session.BootstrapSession(store)
+	require.NoError(t, err)
+	sm := testkit.MockSessionManager{}
+	dom.InfoSyncer().SetSessionManager(&sm)
+	tk := testkit.NewTestKit(t, store)
+	// set it to default value.
+	tk.MustExec(fmt.Sprintf("set global innodb_lock_wait_timeout = %d", vardef.DefInnodbLockWaitTimeout))
+	tk.MustExec("use test")
+
+	if !option.retainData {
+		tk.MustExec("delete from mysql.tidb_global_task;")
+		tk.MustExec("delete from mysql.tidb_background_subtask;")
+		tk.MustExec("delete from mysql.tidb_ddl_job;")
```

Comment:
why run those cleanup here?

Reply:
So we can make sure there is no partial DDL/DXF state from the previous real tikv tests.